clippy_lints/src/undocumented_unsafe_blocks.rs

```diff
@@ -639,29 +615,42 @@ fn get_body_search_span(cx: &LateContext<'_>) -> Option<Span> {
     let body = cx.enclosing_body?;
     let map = cx.tcx.hir();
     let mut span = map.body(body).value.span;
+    let mut maybe_global_var = false;
     for (_, node) in map.parent_iter(body.hir_id) {
         match node {
-            Node::Expr(e) => span = e.span,
+            Node::Expr(e) => {
+                span = e.span;
+                // Note: setting this to `false` is to making sure a "in-function defined"
+                // const/static variable not mistakenly processed as global variable,
+                // since global var doesn't have an `Expr` parent as its parent???
```

Comment:
I'm 70% sure of this... so, could someone confirm this for me? 😆

Reply:
I *think* this is because a `const`'s initializer is a body, and when getting a body's `value`'s parent it'll get wherever the body's used; see the HIR for `const A: u32 = { 0 }`: https://rust.godbolt.org/z/3cY4vaq6T

So this is getting the `Expr` every other time, yet here, it's getting the `Const`/`Static`, afaik?

Reply:
wait a minute, I think unintentionally increase the search span for other scenarios, let me see if I can find a better solution~